### PR TITLE
Inline editing of Text component breaks SPA state update

### PR DESCRIPTION
**Pull request by habansal**
* Added a data attribute on the container div for the Text component html
* Used that attribute as a selector for the inlineEditing config - `editElementQuery` so that AEM InlineTextEditor uses that div as the RTE container and doesn't delete it
* Also, did the same for the Angular component

### CODE REVIEW

File: react-app/src/components/Text.js [hunk 1]
```diff
@@ -39,7 +39,7 @@ const TextEditConfig = {
 class Text extends Component {
 
     get richTextContent() {
-        return <div id={extractModelId(this.props.cqPath)} dangerouslySetInnerHTML={{__html:  this.props.text}}/>;
+        return <div id={extractModelId(this.props.cqPath)} data-rte-editelement="true" dangerouslySetInnerHTML={{__html:  this.props.text}}/>;
```
Comment (pfauchere) [inline react-app/src/components/Text.js]: the boolean may not be needed: `data-rte-editelement` could be enough

Comment (pfauchere) [reply react-app/src/components/Text.js]:
```suggestion
        return <div id={extractModelId(this.props.cqPath)} data-rte-editelement dangerouslySetInnerHTML={{__html:  this.props.text}}/>;
```

**Review by pfauchere — DISMISSED** [review]

**Comment by pfauchere** [comment]
Let's find out why the following error occurs

`Error: Could not find or load main class org.apache.maven.surefire.booter.ForkedBooter`

**Review by fistakus — APPROVED** [review]
Would prefer to avoid hard-coding it in 2 places (especially in the fe component) and taking i.e. from props (passed there by the libraries provided) but seems that this is a nice and easy solution for the problem so +1

**Pull request merged**